src/parser/priest/discipline/normalizers/AtonementSuccessiveDamage.ts

```diff
@@ -1,35 +1,35 @@
-import { EventType } from 'parser/core/Events';
-import EventsNormalizer from "parser/core/EventsNormalizer";
+import { DamageEvent, Event, EventType, HealEvent } from 'parser/core/Events';
+import EventsNormalizer from 'parser/core/EventsNormalizer';
 
-import isAtonement from "../modules/core/isAtonement";
-import { ATONEMENT_DAMAGE_SOURCES } from "../constants";
+import isAtonement from '../modules/core/isAtonement';
+import { ATONEMENT_DAMAGE_SOURCES } from '../constants';
 
 class AtonementSuccessiveDamage extends EventsNormalizer {
-  normalize(events) {
-    const fixedEvents = [];
-    const _damageEventIndexes = [];
+  normalize(events: Array<Event<any>>) {
+    const fixedEvents: Array<Event<any>> = [];
+    const _damageEventIndexes: number[] = [];
 
-    let _encounteredTargetIDs = [];
+    let _encounteredTargetIDs: number[] = [];
 
-    events.forEach((event, eventIndex) => {
+    events.forEach((event: Event<any>, eventIndex) => {
       fixedEvents.push(event);
 
       if (
         event.type === EventType.Damage &&
-        event.sourceIsFriendly &&
-        !event.targetIsFriendly &&
-        ATONEMENT_DAMAGE_SOURCES[event.ability.guid]
+        (event as DamageEvent).sourceIsFriendly &&
+        !(event as DamageEvent).targetIsFriendly &&
+        ATONEMENT_DAMAGE_SOURCES[(event as DamageEvent).ability.guid]
       ) {
         _damageEventIndexes.push(eventIndex);
         _encounteredTargetIDs = [];
         return;
       }
 
-      if (event.type === EventType.Heal && isAtonement(event)) {
+      if (event.type === EventType.Heal && isAtonement((event as HealEvent))) {
```

Comment:
Just as a comment, I am going to PR a fix for this so that you don't have to cast this every time. I'll @ you to see the fix

Reply:
I've put this fix in, AnyEvent now doesn't require the casts in this file (`as XEvent` should all be able to remove)